pypeit/reduce.py

```diff
@@ -645,60 +662,119 @@ def load_skyregions(self, skymask_init):
+                mode = "global"
+            elif mode not in ["local", "global"]:
+                msgs.warn("mode must be 'global' or 'local'. Assuming 'global'.")
+                mode = "global"
+
+            # Prepare a list of slit spectra, if required.
+            if mode == "global":
+                # TODO :: Need to think about spatial flexure - is the appropriate spatial flexure already included in trace_spat via left/right slits?
+                trace_spat = 0.5 * (self.slits_left + self.slits_right)#/(self.slits.nspat-1)
+                trace_spec = np.arange(self.slits.nspec)#/(self.slits.nspec-1)
+                slitSpecs = []
+                for ss in range(self.slits.nslits):
+                    if not gd_slits[ss]:
+                        slitSpecs.append(None)
+                        continue
+                    slit_spat = self.slits.spat_id[ss]
+                    mask = (self.slitmask == slit_spat)
+                    box_denom = moment1d(self.waveimg * mask > 0.0, trace_spat[:, ss], 2, row=trace_spec)[0]
+                    wghts = (box_denom + (box_denom == 0.0))
+                    slit_sky = moment1d(self.global_sky * mask, trace_spat[:, ss], 2, row=trace_spec)[0] / wghts
+                    # Denom is computed in case the trace goes off the edge of the image
+                    slit_wave = moment1d(self.waveimg * mask, trace_spat[:, ss], 2, row=trace_spec)[0] / wghts
+                    slitSpecs.append(xspectrum1d.XSpectrum1D.from_tuple((slit_wave, slit_sky)))
+
+            # Measure flexure
+            flex_list = flexure.spec_flexure_slit(self.slits, self.slits.slitord_id, self.reduce_bpm,
+                                                  self.par['flexure']['spectrum'],
+                                                  method=self.par['flexure']['spec_method'],
+                                                  mxshft=self.par['flexure']['spec_maxshift'],
+                                                  specobjs=specObjs, slitspecs=slitSpecs)
+            if mode == "global":
+                # Store the slit shifts that were applied to each slit
+                # These corrections are later needed so the headers of the specobjs contains the total spectral flexure
+                self.slitshift = np.zeros(self.slits.nslits)
+                for islit in range(self.slits.nslits):
+                    if (not gd_slits[islit]) or len(flex_list[islit]['shift']) == 0:
+                        continue
+                    self.slitshift[islit] = flex_list[islit]['shift'][0]
+                # Apply the new wavelength solution
+                msgs.info("Regenerating wavelength image")
+                self.waveimg = self.wv_calib.build_waveimg(self.tilts, self.slits,
+                                                           spat_flexure=self.spat_flexure_shift,
+                                                           spec_flexure=self.slitshift)
+
```

Comment:
So if mode == 'local' where is the flexure being applied? Is this done internally to the specobjs? I'm struggling with the logical flow here.